internal/cmd/marketplace/list.go

```diff
@@ -0,0 +1,94 @@
+// Copyright Mia srl
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package marketplace
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/mia-platform/miactl/internal/client"
+	"github.com/mia-platform/miactl/internal/clioptions"
+	"github.com/mia-platform/miactl/internal/resources"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+)
+
+const (
+	listMarketplaceEndpoint = "/api/backend/marketplace/"
+)
+
+// ListCmd return a new cobra command for listing marketplace items
+func ListCmd(options *clioptions.CLIOptions) *cobra.Command {
+	return &cobra.Command{
+		Use:   "list",
+		Short: "List marketplace items",
+		Long: `List the marketplace items that the current user can access.
+
+Mia-Platform Marketplace contains several code resources which help develop any architecture faster.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			restConfig, err := options.ToRESTConfig()
+			cobra.CheckErr(err)
+			client, err := client.APIClientForConfig(restConfig)
+			cobra.CheckErr(err)
+			return listMarketplaceItems(client, restConfig.CompanyID)
+		},
+	}
+}
+
+// listMarketplaceItems retrieves the marketplace items belonging to the current context
+func listMarketplaceItems(client *client.APIClient, companyID string) error {
+	req := client.
+		Get().
+		APIPath(listMarketplaceEndpoint)
+
+	if len(companyID) != 0 {
+		req.SetParam("tenantId", companyID)
+	}
+	// execute the request
+	resp, err := req.Do(context.Background())
```

Comment:
I'd be better maybe to split the API invocation from the "table rendering" in two functions, this way we could re-use the list api in the future

Reply:
Now the API invocation is in a dedicated function 👍🏼